File: leetcode/number-of-good-pairs.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        int size = nums.size();
        int count = 0;
        
        for(int i = 1; i < size; ++i) {
            for(int j = 0; j < i; ++j) {
                if (nums[i] == nums[j]) {
                    ++count;
                }
            }
        }
        
        return count;
    }
};

File: leetcode/number-of-good-pairs.py
from typing import List

class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        length = len(nums)
        count = 0

        for i,num in enumerate(nums):
            for j in range(i+1, length):
                if num == nums[j]:
                    count += 1
        return count
